### PR TITLE
Support attributes on array members

### PR DESCRIPTION
**Pull request by etaoins**
Array members are allowed to have attributes such as `#[cfg]`.

### CODE REVIEW

File: crates/ra_parser/src/grammar/expressions/atom.rs [hunk 1]
```diff
@@ -170,8 +170,28 @@ fn array_expr(p: &mut Parser) -> CompletedMarker {
...
     expr(p);
     if p.eat(T![;]) {
+        if first_member_has_attrs {
+            // test_err array_length_attributes
+            // pub const A: &[i64] = &[
+            //   #[cfg(test)]
+            //   1;
+            //   2,
+            // ];
+            p.error("removing an expression is not supported in this position");
```
Comment (matklad) [inline crates/ra_parser/src/grammar/expressions/atom.rs]: Should this be a *parse* error though? The error message sounds like it's from macro-expansion and cfg processing. 

Looks like we should just allow attributes on all expressions?

Comment (etaoins) [reply crates/ra_parser/src/grammar/expressions/atom.rs]: Ah yes, you're right. It looks like the `rustc` parser allows arbitrary attributes here. I'll remove the error.

**Comment by matklad** [comment]
Some more experiments:

This compiles on stable:

```rust
fn main() {
    [#[allow(unused)] (), (), ()];
}
```

This fails with a nightly-feature error (attributes on expressions are experimental. (see issue #15701)):

```rust
fn main() {
    [#[allow(unused)] (); 92];
}
```

The fact that the first case parses without feature flags seems like a bug in feature-gating to me.

I guess what we should is to allow attributes on array expressions unconditionally.

**Comment by matklad** [comment]
bors r+

Thanks!

**Pull request merged**